### PR TITLE
week_15_solutions

### PR DESCRIPTION
**Pull request by mustafacihat**

### CODE REVIEW

**Review by sulkemref — APPROVED** [review]
Good!

File: src/main/java/week_by_week/week15/LinkedListCycleDetection.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    public ListNode detectCycleSet(ListNode head) {
+
+        Set<ListNode> set = new HashSet<>();
+
+        ListNode current = head;
+
+        while(current != null){
+            if(!set.add(current)){
+                return current;
```
Comment (Rikhsiboev) [inline src/main/java/week_by_week/week15/LinkedListCycleDetection.java:19]: ```suggestion  no need to change nothing i am just practising from intlj how to commint and bring suggestion or commit    ```
public ListNode detectCycleSet(ListNode head) {

        Set<ListNode> set = new HashSet<>();

        ListNode current = head;

        while(current != null){
            if(!set.add(current)){
                return current;

```  public ListNode detectCycleSet(ListNode head) {

        Set<ListNode> set = new HashSet<>();

        ListNode current = head;

        while(current != null){
            if(!set.add(current)){
                return current;```

**Review by Rikhsiboev — CHANGES_REQUESTED** [review]
no need any action i am just leaning and practicing #51

File: src/main/java/week_by_week/week15/LinkedListCycleDetection.java (hunk 1 above)
Comment (mustafacihat) [reply src/main/java/week_by_week/week15/LinkedListCycleDetection.java:19]: I got your point thanks Bobur

**Review by mustafacihat — COMMENTED** [review]
thank you

**Pull request merged**